js/web/editor.vue

```diff
@@ -0,0 +1,208 @@
+                    $gettext(
+                        'Please enter filename to which this document should be stored.'
+                    )
+                "
+                :input-value="fileName"
+                class="oc-mb-l uk-position-absolute"
+                v-on:cancel="cancelSaveAs"
+                v-on:confirm="confirmSaveAs"
+                :hidden="hideSaveAs"
+            />
+            <div id="mainContainer" />
+        </div>
+    </main>
+</template>
+
+<script>
+import { mapActions, mapGetters } from "vuex";
+import axios from "axios";
+
+export default {
+    data: () => ({
+        mode: "edit",
+        fileId: null,
+        filePath: null,
+        fileName: null,
+        hideSaveAs: true,
+    }),
+
+    created() {
+        this.mode = this.$route.params.mode;
+        this.fileId = this.$route.params.fileId;
+        this.filePath = this.$route.params.filePath;
+    },
+
+    methods: {
+        ...mapActions(["showMessage"]),
+
+        messageDisplay(desc, status = "danger", title = "") {
+            this.showMessage({
+                title: title,
+                desc: desc,
+                status: status,
+                autoClose: {
+                    enabled: true,
+                },
+            });
+        },
+
+        cancelSaveAs() {
+            this.hideSaveAs = true;
+        },
+
+        confirmSaveAs(value) {
+            if (value) {
+                let frame = document.getElementById("loleafletframe");
+                this.WOPIPostMessage(frame, "Action_SaveAs", {
+                    Filename: value,
+                    Notify: true,
+                });
+                this.hideSaveAs = true;
+            }
+        },
+
+        onRequestClose() {
+            let params = { item: null };
+            if (this.currentFolder) {
+                params.item = this.currentFolder.path;
+            }
+
+            this.$router.push({ name: "files-personal", params });
+        },
+
+        WOPIPostMessage(iframe, msgId, values) {
+            if (iframe) {
+                var msg = {
+                    MessageId: msgId,
+                    SendTime: Date.now(),
+                    Values: values,
+                };
+
+                iframe.contentWindow.postMessage(JSON.stringify(msg), "*");
+            }
+        },
+
+        // where we get document url from owncloud api
+        async getDocumentFileInfo() {
+            return axios({
+                method: "GET",
+                url:
+                    this.configuration.server +
+                    "index.php/apps/richdocuments/ajax/documents/index/" +
+                    this.fileId,
+                headers: {
+                    Authorization: "Bearer " + this.getToken,
+                },
+            }).then((response) => {
+                return response.data;
+            });
+        },
+
+        generateWOPISrc(documentInfo) {
+            var documentId =
```

Comment:
this can be simplified to 
```js
const documentId = [documentInfo.fileId, documentInfo.instanceId, documentInfo.version, documentInfo.sessionId].join('_')
```